apps/opik-backend/src/main/java/com/comet/opik/utils/ValidationUtils.java

```diff
@@ -7,7 +7,7 @@
 
 public class ValidationUtils {
 
-    public static final String NULL_OR_NOT_BLANK = "^(?!\\s*$).+";
+    public static final String NULL_OR_NOT_BLANK = "(?s)^\\s*(\\S.*\\S|\\S)\\s*$";
```

Comment:
You should add a javadoc explaining the regex. It's becoming too complicated.

Reply:
Done